src/Cardano/BM/Trace.lhs

```diff
@@ -105,15 +108,26 @@ The context name is created and checked that its size is below a limit
 The minimum severity that a log message must be labelled with is looked up in
 the configuration and recalculated.
 \begin{code}
-modifyName :: MonadIO m => LoggerName -> Trace m -> m (Trace m)
-modifyName name (ctx, trace) = return ( ctx { loggerName = name }, trace )
+modifyName :: MonadIO m => (LoggerName -> LoggerName) -> Trace m -> m (Trace m)
+modifyName f (ctx, basetrace0) =
+    let basetrace = modifyNameBase f basetrace0
+    in
+    return (ctx, basetrace)
+
+modifyNameBase
+    :: (LoggerName -> LoggerName)
+    -> TraceNamed m
+    -> TraceNamed m
+modifyNameBase k = contramap f
+  where
+    f (LogNamed name item) = LogNamed (((T.take 80) . k) name) item
```

Comment:
Not working as expected.

Reply:
How so?

I wouldn't recommend putting a `take 80` here. I'd think that name truncation should be a switchboard policy.

Reply:
```
*Cardano.BM.Trace Cardano.BM.Data.Aggregated> lo <-  LogObject <$> (mkLOMeta Debug) <*> pure (LogValue "rr" (PureI 1))

*Cardano.BM.Trace Cardano.BM.Data.Aggregated> tr = modifyNameBase (\a -> "new" <> a) stdoutTrace  
*Cardano.BM.Trace Cardano.BM.Data.Aggregated> BaseTrace.traceWith (named tr) lo
new :: {"tag":"LogValue","contents":["rr",{"tag":"PureI","contents":1}]}

*Cardano.BM.Trace Cardano.BM.Data.Aggregated> tr2 = modifyNameBase (\a -> ".york" <> a) tr
*Cardano.BM.Trace Cardano.BM.Data.Aggregated> BaseTrace.traceWith (named tr2) lo
new.york :: {"tag":"LogValue","contents":["rr",{"tag":"PureI","contents":1}]}

*Cardano.BM.Trace Cardano.BM.Data.Aggregated> tr3 = modifyNameBase (\a -> ".city" <> a) tr2
*Cardano.BM.Trace Cardano.BM.Data.Aggregated> BaseTrace.traceWith (named tr3) lo
new.york.city :: {"tag":"LogValue","contents":["rr",{"tag":"PureI","contents":1}]}

*Cardano.BM.Trace Cardano.BM.Data.Aggregated> tr4 = modifyNameBase (const "OTHER" ) tr3
*Cardano.BM.Trace Cardano.BM.Data.Aggregated> BaseTrace.traceWith (named tr4) lo
new.york.cityOTHER :: {"tag":"LogValue","contents":["rr",{"tag":"PureI","contents":1}]}
```

The last `const` is not working as expected I guess it just throws the empty `String`.
The order of the new and old names must be switched.

Reply:
Weird, that doesn't look right. It should be `new`, then `.yorknew`, then .city.yorknew`, no? `modifyNameBase (const "OTHER") tr3 really should forget the `new.york.city` part.

Reply:
Oh never mind, I had it backwards. `named tr4` will put in an `mempty`, which will be changed to `"OTHER"`, which will then get the others prefixed

Reply:
Yeah, so everything here is fine, it's just that once you `modifyBaseName`, you can't ever remove that prefix from the resulting trace. Instead of trying to overwrite with `const "OTHER"`, just use the original trace `stdoutTrace` from which `tr4` is ultimately derived anyway.

Reply:
A typical usage pattern for this (as far as I'm aware the only one) is to take a named trace _for which you don't care what the name is_ and to then append your own suffix.